app/shop.py

```diff
@@ -0,0 +1,16 @@
+from typing import Tuple
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: Tuple[int, int],
+            products: dict

```

Comment:
Consider using a more specific type for 'products' instead of 'dict'. You can use 'Dict[str, int]' where str represents the product name and int represents the quantity or price (depending on your implementation).